protocol/src/main/java/org/geysermc/mcprotocollib/protocol/packet/status/clientbound/ClientboundStatusResponsePacket.java

```diff
@@ -31,13 +31,25 @@ public class ClientboundStatusResponsePacket implements MinecraftPacket {
     // vanilla behavior falls back to false if the field was not sent
     private static final boolean ENFORCES_SECURE_CHAT_DEFAULT = false;
 
-    private final @NonNull ServerStatusInfo info;
+    private final @NonNull JsonObject data;
+
+    public ClientboundStatusResponsePacket(ServerStatusInfo info) {
+        this(toJson(info));
+    }
 
     public ClientboundStatusResponsePacket(ByteBuf in, MinecraftCodecHelper helper) {
-        JsonObject obj = new Gson().fromJson(helper.readString(in), JsonObject.class);
-        JsonElement desc = obj.get("description");
+        data = new Gson().fromJson(helper.readString(in), JsonObject.class);
+    }
+
+    @Override
+    public void serialize(ByteBuf out, MinecraftCodecHelper helper) {
+        helper.writeString(out, data.toString());
+    }
+
+    public ServerStatusInfo getInfo() {
```

Comment:
I understand this was named as so to avoid breaking changes, but I'm still gonna propose naming it something like `parseInfo` to indicate that it shouldn't be called repeatedly